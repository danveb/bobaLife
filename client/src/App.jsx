import { useState } from "react";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom"; 
import Home from "./pages/Home"; 
import About from "./pages/About"; 
import Login from "./pages/Login"; 
import Register from "./pages/Register"; 
import { ToastContainer } from "react-toastify"; 
import "react-toastify/dist/ReactToastify.css"; 

const App = () => {
    const [menuOpen, setMenuOpen] = useState(false); 
    
    return (
        <>
            <Router>
                <Routes>
                    <Route path="/" element={<Home menuOpen={menuOpen} setMenuOpen={setMenuOpen} /> } />
                    <Route path="/about" element={<About menuOpen={menuOpen} setMenuOpen={setMenuOpen} /> } />
                    <Route path="/login" element={<Login menuOpen={menuOpen} setMenuOpen={setMenuOpen}/> } />
                    <Route path="/register" element={<Register menuOpen={menuOpen} setMenuOpen={setMenuOpen}/> } />
                </Routes>
            </Router>
            <ToastContainer />
        </>
    )
}

export default App 