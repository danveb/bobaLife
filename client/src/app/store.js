import { combineReducers, configureStore } from "@reduxjs/toolkit"; 
import authReducer from "../redux/auth/authSlice"; 
import productsReducer from "../redux/products/productsSlice"; 
import cartReducer from "../redux/cart/cartSlice"; 
import {
    persistStore,
    persistReducer,
    FLUSH,
    REHYDRATE,
    PAUSE,
    PERSIST,
    PURGE,
    REGISTER,
} from 'redux-persist';
import storage from "redux-persist/lib/storage"; 

// Perhaps combine all reducers and persist? 
// even persist user so we don't save user data in localStorage?
// combineReducers from rootReducer
const rootReducer = combineReducers({
    auth: authReducer, 
    products: productsReducer, 
    cart: cartReducer, 
}); 

const persistConfig = {
    key: 'root',
    storage,
}
  
// const persistedReducer = persistReducer(persistConfig, cartReducer)
const persistedReducer = persistReducer(persistConfig, rootReducer)

export const store = configureStore({
    // reducer: {
    //     auth: authReducer, 
    //     products: productsReducer, 
    //     cart: persistedReducer, 
    // },
    reducer: persistedReducer, 
    middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
            serializableCheck: {
                ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
            },
        }),
}); 

export const persistor = persistStore(store) 