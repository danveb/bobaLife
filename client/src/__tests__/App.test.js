import { render } from "@testing-library/react"; 
import { store } from "../app/store"; 
import { Provider } from "react-redux";
import App from "../App"; 

// smoke test
test("renders without crashing", () => {
    render(
        <Provider store={store}>
            <App />
        </Provider>
    ); 
}); 

// snapshot test
test("matches snapshot", () => {
    const { asFragment } = render(
        <Provider store={store}>
            <App />
        </Provider>
    ); 
    expect(asFragment()).toMatchSnapshot(); 
});