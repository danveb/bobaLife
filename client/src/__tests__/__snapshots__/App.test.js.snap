// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <div
      class="nav-wrapper false"
    >
      <div
        class="nav-left"
      >
        <a
          class="logo"
          href="/"
        >
          bobaLife
        </a>
      </div>
      <div
        class="nav-center"
      >
        <ul
          class="nav-links"
        >
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="/drinks"
            >
              Drinks
            </a>
          </li>
          <li>
            <a
              href="/shop"
            >
              Shop
            </a>
          </li>
        </ul>
      </div>
      <div
        class="nav-right"
      >
        <a
          href="/login"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AccountCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
            />
          </svg>
        </a>
        <a
          href="/cart"
        >
          <span
            class="MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ShoppingBasketIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m17.21 9-4.38-6.56c-.19-.28-.51-.42-.83-.42-.32 0-.64.14-.83.43L6.79 9H2c-.55 0-1 .45-1 1 0 .09.01.18.04.27l2.54 9.27c.23.84 1 1.46 1.92 1.46h13c.92 0 1.69-.62 1.93-1.46l2.54-9.27L23 10c0-.55-.45-1-1-1h-4.79zM9 9l3-4.4L15 9H9zm3 8c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
              />
            </svg>
            <span
              class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary BaseBadge-badge BaseBadge-invisible css-1pi4uwz-MuiBadge-badge"
            />
          </span>
        </a>
        <div
          class="hamburger"
        >
          <span />
          <span />
          <span />
        </div>
      </div>
    </div>
  </div>
  <div
    class="menu false"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/drinks"
        >
          Drinks
        </a>
      </li>
      <li>
        <a
          href="/shop"
        >
          Shop
        </a>
      </li>
    </ul>
  </div>
  <div
    class="sc-bczRLJ cxfrMe"
  >
    <div
      class="sc-crXcEl bjHRtD"
    >
      <div
        class="sc-evZas kjfDWn"
      >
        <div
          class="sc-breuTD hemlwj"
          id="1"
        >
          <h1
            class="sc-ksZaOG dqsHfU"
          >
            Welcome to bobaLife
          </h1>
          <p
            class="sc-hAZoDl dGRemi"
          >
            Summer | 2022
          </p>
        </div>
      </div>
    </div>
    <div
      class="sc-gsnTZi hLOdTa"
    >
      <div
        class="sc-dkzDqf nuaNw"
      >
        <div
          class="sc-hKMtZM dYVNGz"
        >
          <div
            class="sc-eCYdqJ jWyszA hero-card"
          >
            <img
              alt="shop"
              class="sc-jSMfEi jDbNbV"
              src="shop.webp"
            />
          </div>
          <div
            class="sc-eCYdqJ jWyszA hero-card"
          >
            <img
              alt="boba"
              class="sc-jSMfEi jDbNbV"
              src="boba.webp"
            />
          </div>
          <div
            class="sc-eCYdqJ jWyszA hero-card"
          >
            <img
              alt="matcha"
              class="sc-jSMfEi jDbNbV"
              src="matcha.webp"
            />
          </div>
        </div>
      </div>
      <div
        class="sc-gKXOVf gGlsUt"
      >
        <div
          class="sc-iBkjds ECrvz"
        >
          <h1
            class="sc-ftvSup kFqKuM"
          >
            let's #bobaLife
          </h1>
          <img
            alt="milk tea"
            class="sc-jSMfEi sc-papXJ jDbNbV hVIVWn"
            src="boba2.webp"
          />
        </div>
        <div
          class="sc-jqUVSM jkcOdD"
        >
          <h3
            class="sc-kDDrLX hjYTIv"
          >
            Why bobaLife?
          </h3>
          <p
            class="sc-iqcoie gOFMmz"
          >
            We have only one obsession, and you guessed it right: boba. Some call it boba, some call it bubble tea, but we all know this word is interchangeable. Life is so short and we are just passionate to share boba to the world. This is exactly what makes us who we are. We want to welcome you to a friendly atmosphere delicately curated to serve you a world-class boba on every single cup. 
            <span>
              Welcome to bobaLife.
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="sc-fnykZs hhRTuM"
  >
    <div
      class="sc-fEOsli cUluRE"
    >
      <div
        class="sc-bjUoiL eNfzxb"
      >
        <h3
          class="sc-idiyUo dwyRkM"
        >
          Contact Us
        </h3>
        <div
          class="sc-dIouRR gbFrDQ"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LanguageIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
            />
          </svg>
          1502 N Milwaukee Ave, Chicago, IL 60622
        </div>
        <div
          class="sc-dIouRR gbFrDQ"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="WhatsAppIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.75 13.96c.25.13.41.2.46.3.06.11.04.61-.21 1.18-.2.56-1.24 1.1-1.7 1.12-.46.02-.47.36-2.96-.73-2.49-1.09-3.99-3.75-4.11-3.92-.12-.17-.96-1.38-.92-2.61.05-1.22.69-1.8.95-2.04.24-.26.51-.29.68-.26h.47c.15 0 .36-.06.55.45l.69 1.87c.06.13.1.28.01.44l-.27.41-.39.42c-.12.12-.26.25-.12.5.12.26.62 1.09 1.32 1.78.91.88 1.71 1.17 1.95 1.3.24.14.39.12.54-.04l.81-.94c.19-.25.35-.19.58-.11l1.67.88M12 2a10 10 0 0 1 10 10 10 10 0 0 1-10 10c-1.97 0-3.8-.57-5.35-1.55L2 22l1.55-4.65A9.969 9.969 0 0 1 2 12 10 10 0 0 1 12 2m0 2a8 8 0 0 0-8 8c0 1.72.54 3.31 1.46 4.61L4.5 19.5l2.89-.96A7.95 7.95 0 0 0 12 20a8 8 0 0 0 8-8 8 8 0 0 0-8-8z"
            />
          </svg>
          +1 (312) 846-9730
        </div>
        <div
          class="sc-dIouRR gbFrDQ"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SendIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
            />
          </svg>
          contact@bobalife.com
        </div>
        <div
          class="sc-dIouRR gbFrDQ"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CreditCardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zm0 14H4v-6h16v6zm0-10H4V6h16v2z"
            />
          </svg>
          Contactless payments accepted
        </div>
      </div>
      <div
        class="sc-hHLeRK eKEfHS"
      >
        <h3
          class="sc-idiyUo dwyRkM"
        >
          Our Hours
        </h3>
        <div
          class="sc-dIouRR gbFrDQ"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="QueryBuilderIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
            <path
              d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
          Every Day @ 11:00 - 21:00
        </div>
      </div>
      <div
        class="sc-dmRaPn hlpnwC"
      >
        <h3
          class="sc-idiyUo dwyRkM"
        >
          bobaLife
        </h3>
        <p
          class="sc-kgflAQ hICsIC"
        >
          Proud to serve a world-class boba carefully crafted by our baristas.
        </p>
        <h5
          class="sc-fLlhyt emGFwR"
        >
          © All rights reserved – Danny Bae
        </h5>
        <div
          class="sc-bBrHrO kKjHNB"
        >
          <a
            class="sc-ivTmOn cFEvfT"
            href="https://www.github.com/danveb"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="GitHubIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
